src/patternfly/components/Select/select.scss

```diff
@@ -380,44 +407,86 @@
   &:hover,
   &:focus,
   &.pf-m-focus {
+    --pf-c-select__menu-item--m-action--Color: var(--pf-c-select__menu-item--hover__menu-item--m-action--Color);
+
     text-decoration: none;
   }
 
-  &:hover {
-    background-color: var(--pf-c-select__menu-item--hover--BackgroundColor);
+  @at-root .pf-c-select__menu-wrapper,
+  & {
+    &:hover {
+      background-color: var(--pf-c-select__menu-item--hover--BackgroundColor);
+    }
   }
 
+  @at-root .pf-c-select__menu-wrapper:focus-within,
   &:focus,
   &.pf-m-focus {
     position: relative;
     background-color: var(--pf-c-select__menu-item--focus--BackgroundColor);
   }
 
-  &:disabled,
-  &.pf-m-disabled {
-    color: var(--pf-c-select__menu-item--disabled--Color);
-    pointer-events: none;
-    background-color: var(--pf-c-select__menu-item--disabled--BackgroundColor);
+  &.pf-m-link {
```

Comment:
That's for the main link part of the item. With the favorites feature, instead of having

```
li
  a/button.pf-c-select__menu-item
    // item contents
```

we have

```
li.pf-c-select__menu-wrapper
  a/button.pf-c-select__menu-item.pf-m-linik // menu item contents
  button.pf-c-select__menu-item.pf-m-action // favorites
```